### DISCUSSION TITLE
Question about underwater tile culling

### DISCUSSION THREAD

**Discussion by Revmatek** [crest-ocean/crest#1134] [General]
@daleeidd 

Occasionally I see this underwater:
![image](https://github.com/wave-harmonic/crest/assets/114962753/7503be97-7741-4256-8cc9-185ffd65c453)

This seems to be the ocean tiles culling, it is directly affected by the far plane multiplier on the underwater renderer, lowering this value seems to mitigate it but not completely eliminate. Is there some other setting I am unaware of? Is this a bug? I have the far plane multiplier set very lower to avoid stuff like this, but setting it to 0 does not eliminate it, nor does that seem like the correct solution.

This seems to happen in the main scene when the far plane multiplier has higher values (which may be normal).

**Comment by magenta404** [comment]
I am a bit confused about that image. Is it suppose to be really tall? Can you annotate or explain the issue you're seeing?

**Reply by Revmatek** [reply]
This is a screenshot of an underwater pov looking up. The rest of the screen is black because it is deep underwater. When you are deep underwater and turn the camera tiles on the edge of the screen start to disappear.

**Comment by Revmatek** [comment]
Here is a better show from BoatWakesScene:
<img width="1890" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/4e97f3b9-d647-4b91-ad88-e2ace606abbd">
<br>
On the left side you can see a tile culled / missing
<img width="45" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/d6070262-3fdb-47b2-9ddb-89a9ccc6640f">

In the example scenes this problem goes away if you decrease 
<img width="437" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/b92c7237-485a-4b0f-9220-6fc49b203cb7">

I already have this setting pretty low in my project and it does not solve the problem some I am trying to figure out if this is a bug or I am missing / forgetting some other key setting that can affect this.

**Reply by magenta404** [reply]
Is this reproducible with Dynamic Waves disabled? Seems related to #849.

**Comment by Revmatek** [comment]
While attempting to reproduce a separate issue I noticed this can also happen when zooming in an out above water:

<img width="1911" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/24ed20b6-90e7-4850-b563-855da1598b6a">
<img width="105" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/4ab70ede-101b-4583-9ba1-576e13f68f25">

**Reply by magenta404** [reply]
Is this reproducible with Dynamic Waves disabled? Seems related to #849.

**Reply by Revmatek** [reply]
It still happens if I disabled "Create Dynamic Wave Sim". It could be related, I assume it will happen whenever its near the edge of the POV/Camera and the conditions are met (whatever those may be).

It does disappear when lowering the LOD to unreasonable levels like 1-3. I think its is based solely on perspective and camera, does not really seem related to anything but the tiles / tile culling.

**Reply by Revmatek** [reply]
@daleeidd 
My original issue is caused by having the far clip plane high underwater on the Camera itself.

Example:
<img width="525" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/ee5f3bfb-71ff-46e5-9577-8ad88992aa36">

Having a far clipping plane is needed for flying at high altitude and seeing the ocean without the skybox / background becoming visible. This problem seems fixable on my end by simply lowering the far clip plane underwater when:
<img width="321" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/fa44447d-37a6-4618-a909-515c6d3d5477">
<img width="528" alt="image" src="https://github.com/wave-harmonic/crest/assets/114962753/02b0d94a-3917-4883-a2fd-2e9d49d0e981">


Not sure there is a bug here, definitely inaccuracy culling underwater tiles / water fog with further view distances.

**Comment by magenta404** [comment]
Solved in 4.20.
